From: Kurt Hindenburg <kurt.hindenburg@gmail.com>
Date: Sat, 28 Jan 2017 10:22:52 -0500
Subject: Allow ANSI string terminator to work

Treat "\e]0;<title>\e\\" (set window title) the same way
it interprets "\e]0;<title>\007".

It seems that Konsole does not recognize the standard string terminator
sequence "\e\\" (ESC \). Everything works fine if I use "\007" (BEL).

<http://invisible-island.net/xterm/xterm.faq.html#how2_title>

Thanks to Aniketh Girish anikethgireesh gmail com and Martin Sandsmark
martin sandsmark kde org for patch/review

BUG:231405
REVIEW:129895
---
 src/Vt102Emulation.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Vt102Emulation.cpp b/src/Vt102Emulation.cpp
index a0c73cef..1bde8469 100644
--- a/src/Vt102Emulation.cpp
+++ b/src/Vt102Emulation.cpp
@@ -280,7 +280,7 @@ void Vt102Emulation::initTokenizer()
 #define epe( )     (p >=  3  && s[2] == '!')
 #define egt( )     (p >=  3  && s[2] == '>')
 #define Xpe        (tokenBufferPos >= 2 && tokenBuffer[1] == ']')
-#define Xte        (Xpe      && cc ==  7 )
+#define Xte        (Xpe      && (cc ==  7 || cc == 33))
 #define ces(C)     (cc < 256 && (charClass[cc] & (C)) == (C) && !Xte)
 
 #define CNTL(c) ((c)-'@')
